lib/matplotlib/tests/test_legend.py

```diff
@@ -1450,3 +1451,53 @@ def test_boxplot_labels():
     handles, labels = ax.get_legend_handles_labels()
     assert len(handles) == 0
     assert len(labels) == 0
+
+
+def test_boxplot_legend_labels():
+    # Test that legend entries are generated from boxplots. `label` is the new
+    # parameter for setting legend labels.
+    np.random.seed(19680801)
+    data = np.random.lognormal(size=(68, 4), mean=1.5, sigma=0.2)
+    labels = list('ABCD')
+    colors = ['peachpuff', 'orange', 'tomato', 'plum']
+
+    legend_labels = ['peaches', 'oranges', 'tomatoes', 'plums']
+    fig, axs = plt.subplots(nrows=2,
+                            ncols=3,
+                            figsize=(10, 7),
+                            sharey=True)
+    # This plot will not generate a legend.
+    first = axs[0, 0].boxplot(data, patch_artist=True)
+    with pytest.warns(UserWarning):
+        axs[0, 0].legend()
+    empty_list = axs[0, 0].get_legend_handles_labels()
+    assert not empty_list[0] and not empty_list[1]
+
+    # Testing legend labels and patch passed to legend.
+    second = axs[0, 1].boxplot(data, patch_artist=True, label=legend_labels)
+    for i, v in enumerate(second['boxes']):
+        assert v.get_label() == legend_labels[i]
+
```

Comment:
Probably also good to check, that `legend_labels` are exactly the labels in the legend (not more or less).
```suggestion
handles, labels = axs[0, 1].get_legend_handles_labels()
assert labels == legend_labels

```